packages/web/src/components/withdraw-usdc-modal/WithdrawUSDCModal.tsx

```diff
@@ -43,40 +51,47 @@ const messages = {
       'Your USDC wallet does not have enough funds to cover this transaction.',
     amountTooLow: 'Please withdraw at least $0.01.',
     invalidAddress: 'A valid Solana USDC wallet address is required.',
+    minCashTransfer: 'A minimum of $5 is required for cash withdrawals.',
     pleaseConfirm:
       'Please confirm you have reviewed the details and accept responsibility for any errors resulting in lost funds.'
   }
 }
 
-export const AMOUNT = 'amount'
-export const ADDRESS = 'address'
-export const CONFIRM = 'confirm'
+const WithdrawUSDCFormSchema = (
+  userBalanceCents: number,
+  minWithdrawBalanceCents: number
+) => {
+  const amount = z
+    .number()
+    .lte(userBalanceCents, messages.errors.insufficientBalance)
 
-type WithdrawFormValues = {
-  [AMOUNT]: number
-  [ADDRESS]: string
-  [CONFIRM]: boolean
-}
-
-const WithdrawUSDCFormSchema = (userBalance: number) => {
-  let amount = z.number().lte(userBalance, messages.errors.insufficientBalance)
-  if (userBalance !== 0) {
-    // If user has no balance, don't validate minimum, the form will just be disabled
-    amount = amount.gte(1, messages.errors.amountTooLow)
-  }
-
-  return z.object({
-    [AMOUNT]: amount,
-    [ADDRESS]: z
-      .string()
-      .refine(
-        (value) => isValidSolAddress(value as SolanaWalletAddress),
-        messages.errors.invalidAddress
-      ),
-    [CONFIRM]: z.literal(true, {
-      errorMap: () => ({ message: messages.errors.pleaseConfirm })
+  return z.discriminatedUnion(METHOD, [
+    z.object({
+      [METHOD]: z.literal(WithdrawalMethod.COINFLOW),
+      // If user has no balance, don't validate minimum, the form will just be disabled
+      [AMOUNT]:
+        userBalanceCents !== 0
+          ? amount.gte(minWithdrawBalanceCents, messages.errors.minCashTransfer)
+          : amount
+    }),
+    z.object({
+      [METHOD]: z.literal(WithdrawalMethod.MANUAL_TRANSFER),
+      // If user has no balance, don't validate minimum, the form will just be disabled
+      [AMOUNT]:
+        userBalanceCents !== 0
+          ? amount.gte(1, messages.errors.amountTooLow)
```

Comment:
not your code but magic number

Reply:
Is the suggestion to use `gt(0)` instead? I think the idea is that we don't want sub-cent values. I could move this into a constant?